main.js

```diff
@@ -535,3 +536,71 @@ const createCalendar = (monthLength, weekLength, firstDay, checkIn, checkOut) =>
 };
 
 console.log(createCalendar(daysInMonth, daysInWeek, dayOfWeek, checkInDate, checkOutDate));
+
+//  lesson-8
+//  1.
+const studentsData = [
+  {
+    firstName: 'Василий',
+    lastName: 'Петров',
+    admissionYear: 2019,
+    courseName: 'Java'
+  },
+  {
+    firstName: 'Иван',
+    lastName: 'Иванов',
+    admissionYear: 2018,
+    courseName: 'JavaScript'
+  },
+  {
+    firstName: 'Александр',
+    lastName: 'Федоров',
+    admissionYear: 2017,
+    courseName: 'Python'
+  },
+  {
+    firstName: 'Николай',
+    lastName: 'Петров',
+    admissionYear: 2019,
+    courseName: 'Android'
+  }
+];
+
+class User {
+  constructor(firstName, lastName) {
+    this.firstName = firstName;
+    this.lastName = lastName;
+  }
+
+  get fullName() {
+    return `${this.firstName} ${this.lastName}`;
+  }
+}
+
+class Student extends User {
+  constructor(firstName, lastName, admissionYear, courseName) {
+    super(firstName, lastName);
+    this.admissionYear = admissionYear;
+    this.courseName = courseName;
+  }
+
+  get course() {
+    return new Date().getFullYear() - this.admissionYear;
+  }
+}
+
+class Students {
+  constructor(data) {
+    this.students = data.map(
+      (x) => new Student(x.firstName, x.lastName, x.admissionYear, x.courseName)
+    );
+  }
+
+  getInfo() {
+    this.students.sort((a, b) => a.course - b.course);
+    this.students.forEach((x) => console.log(`${x.fullName} - ${x.courseName}, ${x.course} курс`));
```

Comment:
I would recommend replacing the `x` variable with`student` for a more readable view

Reply:
everything else is good